lib/facade_converter.ts

```diff
@@ -92,19 +95,34 @@ export class FacadeConverter extends base.TranspilerBase {
    * Searches for type references that require extra imports and emits the imports as necessary.
    */
   emitExtraImports(sourceFile: ts.SourceFile) {
-    var hasXmlHttpRequest = (n: ts.Node): boolean => {
-      if (n.kind === ts.SyntaxKind.TypeReference) {
-        var type = base.ident((<ts.TypeReferenceNode>n).typeName);
-        if (type === 'XMLHttpRequest') {
-          return true;
+    var libraries = <ts.Map<string>>{
+      "XMLHttpRequest": "dart:html",
+      "Uint8Array": "dart:typed_arrays",
+      "ArrayBuffer": "dart:typed_arrays"
+    };
+    var emitted: Set = {};
+    this._emitImports(sourceFile, libraries, emitted);
+  }
+
+  private _emitImports(n: ts.Node, libraries: ts.Map<string>, emitted: Set,
```

Comment:
I don't think we typically use underscore naming in ts2dart (it's pure typescript so the private keyword is sufficient) - can you check that and remove if this is the first underscore prefix being introduced?